rabbitmq/consumer.go

```diff
@@ -0,0 +1,113 @@
+package rabbitmq
+
+import (
+	"github.com/palantir/stacktrace"
+
+	"go.uber.org/zap"
+
+	"github.com/sumup-oss/go-pkgs/logger"
+
+	"github.com/streadway/amqp"
+)
+
+func NewRabbitMQConsumer(
+	client *RabbitMQClient,
+	handler Handler,
+	logger logger.StructuredLogger, // decide if we want it, maybe not just return errors
+	metric Metric,
+) *RabbitMQConsumer {
+	return &RabbitMQConsumer{
+		client:  client,
+		done:    make(chan error),
+		handler: handler,
+		logger:  logger,
+		metric:  metric,
+	}
+}
+
+type RabbitMQConsumer struct {
+	client  *RabbitMQClient
+	done    chan error
+	handler Handler
+	logger  logger.StructuredLogger
+	metric  Metric
+}
+
+func (c *RabbitMQConsumer) Run(shutdownChan <-chan struct{}) error {
+
+	go func() {
+		select {
+		case <-shutdownChan:
+			c.logger.Info("Received shutdown signal. Going to close rabbit connections.")
+			c.client.channel.Cancel(c.handler.GetConsumerTag(), false) //this will wait for the consumer to finish
```

Comment:
`false` must be a configurable option for the client as part of the struct.

e.g `mustWaitToFinish`. Then the comment will be redundant.